src/main/java/com/commercetools/http/converter/json/PrettyGsonMessageConverter.java

```diff
@@ -0,0 +1,148 @@
+     *
+     * @see GsonHttpMessageConverter#setJsonPrefix(String)
+     */
+    @Override
+    public void setJsonPrefix(String jsonPrefix) {
+        super.setJsonPrefix(jsonPrefix);
+        this.jsonPrefix = jsonPrefix;
+    }
+
+    /**
+     * Because base {@link GsonHttpMessageConverter#jsonPrefix} is private, but is used in overloaded
+     * {@link #writeInternal(Object, Type, HttpOutputMessage)} - we should copy this value.
+     *
+     * @see GsonHttpMessageConverter#setPrefixJson(boolean)
+     */
+    @Override
+    public void setPrefixJson(boolean prefixJson) {
+        super.setPrefixJson(prefixJson);
+        this.jsonPrefix = (prefixJson ? ")]}', " : null);
+    }
+
+    /**
+     * Allow response JSON pretty print if {@code objectToWrite} is a {@link PrettyFormattedBody} instance with
+     * <code>{@link PrettyFormattedBody#isPretty() isPretty} == true</code>.
+     *
+     * @param objectToWrite if the value is {@link PrettyFormattedBody} instance with
+     *                      <code>{@link PrettyFormattedBody#isPretty() isPretty} == true</code> - use
+     *                      {@link #prettyGson} for output writing. Otherwise use base
+     *                      {@link GsonHttpMessageConverter#writeInternal(Object, Type, HttpOutputMessage)}
+     * @param type          the type of object to write (may be {@code null})
+     * @param outputMessage the HTTP output message to write to
+     * @throws IOException                     in case of I/O errors
+     * @throws HttpMessageNotWritableException in case of conversion errors
+     */
+    @Override
+    protected void writeInternal(@Nullable final Object objectToWrite,
+                                 @Nullable final Type type,
+                                 @Nonnull final HttpOutputMessage outputMessage)
+            throws IOException, HttpMessageNotWritableException {
+
+        // based on: if objectToWrite is PrettyFormattedBody && isPretty == true => use custom formatter
+        // otherwise - use the default base GsonHttpMessageConverter#writeInternal(Object, Type, HttpOutputMessage)
+
+        final Optional<PrettyFormattedBody> prettyFormatted = Optional.ofNullable(objectToWrite)
+                .filter(o -> o instanceof PrettyFormattedBody)
+                .map(o -> (PrettyFormattedBody) objectToWrite);
+
+        final boolean pretty = prettyFormatted.map(PrettyFormattedBody::isPretty).orElse(false);
+        final Object realObject = prettyFormatted.map(PrettyFormattedBody::getBody).orElse(objectToWrite);
+
+        if (pretty) {
+            // this is basically full copy of super.writeInternal(), but with custom (pretty) gson mapper
+            Charset charset = getCharset(outputMessage.getHeaders());
+            OutputStreamWriter writer = new OutputStreamWriter(outputMessage.getBody(), charset);
+            try {
+                if (this.jsonPrefix != null) {
```

Comment:
since it is 

> this is basically full copy of super.writeInternal(), but with custom (pretty) gson mapper

i left it untouched. In other case i really recommend and follow myself to avoid **this** reference.

Reply:
fair enough 😊